src/coreclr/jit/optimizer.cpp

```diff
@@ -2546,12 +2546,7 @@ void Compiler::optFindNaturalLoops()
 
     if (mod)
     {
-        // Need to renumber blocks now since loop canonicalization
-        // depends on it; can defer the rest of fgUpdateChangedFlowGraph()
-        // until after canonicalizing loops. Dominator information is
-        // recorded in terms of block numbers, so flag it invalid.
-        fgDomsComputed = false;
-        fgRenumberBlocks();
+        fgUpdateChangedFlowGraph(FlowGraphUpdates::COMPUTE_DOMS);
```

Comment:
We need to recompute the old dominators here since `optFindAndScaleGeneralLoopBlocks` currently depends on them, but we should be able to get rid of this computation very soon. The current TP regressions are because of cases where we compute the dominators here and then also again after new canonicalization; in those cases only the latter computation is necessary (but it's a little hard to detect, so I just left it in both places for now).